test/integration/controller/admissionchecks/provisioning/provisioning_test.go

```diff
@@ -597,10 +598,38 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 					state := workload.FindAdmissionCheck(updatedWl.Status.AdmissionChecks, ac.Name)
 					g.Expect(state).NotTo(gomega.BeNil())
 					g.Expect(state.State).To(gomega.Equal(kueue.CheckStateReady))
+					g.Expect(state.PodSetUpdates).To(gomega.BeComparableTo([]kueue.PodSetUpdate{
+						{
+							Name: "ps1",
+							Annotations: map[string]string{
+								provisioning.ConsumesAnnotationKey: provReqKey.Name,
+							},
+						},
+						{
+							Name: "ps2",
+							Annotations: map[string]string{
+								provisioning.ConsumesAnnotationKey: provReqKey.Name,
+							},
+						},
+					}))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Check the workload is admitted", func() {
+				util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, updatedWl)
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, wlKey, updatedWl)).To(gomega.Succeed())
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, updatedWl)
 				}, util.Timeout, util.Interval).Should(gomega.Succeed())
 			})
 
 			ginkgo.By("Deleting the provision request", func() {
+				createdRequest := &autoscaling.ProvisioningRequest{
```

Comment:
Can we revert this change to minimize the diff - does not seem related?

Reply:
Still we need to delete the provisioning request, if not set ObjectMeta, there will be an Error like:
```
      resource name may not be empty
      {
          s: "resource name may not be empty",
      }
```

Reply:
Yes, we delete the provisioning request, I was just thinking about the place where we declare the variable.

Reply:
This Error happens both on `k8sClient.Delete(ctx, createdRequest)` and `util.SyncAdmittedConditionForWorkloads(ctx, k8sClient, updatedWl)`.

Reply:
I'm ok with either way.

Reply:
Don't you get the `createdRequest` populated at `k8sClient.Get(ctx, provReqKey, createdRequest)`?

Reply:
I mean, this used to work in the test, right? You were deleting the provisioning request before this PR. I'm just thinking the way it was setup does not need to change,

Reply:
@mimowo We have discussed about `k8sClient.Get(ctx, provReqKey, createdRequest)` in https://github.com/kubernetes-sigs/kueue/pull/1493#discussion_r1431289995, and then we decided not to call api-server .

Reply:
Oops. I misunderstood what you wanted to say.

Reply:
I think the old method had the same number of requests actually, we still call the `Get` at the same places.